src/compiler/emitter.ts

```diff
@@ -7885,5 +7883,22 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                 emitSkipped = writeDeclarationFile(declarationFilePath, sourceFiles, isBundledEmit, host, resolver, emitterDiagnostics) || emitSkipped;
             }
         }
+
+        /**
+         * Write the file names for the emitted files to the console, don't actually compile and emit.
+         * This helps build tools to determine if they should invoke a full compilation.
+         */
+        function emitOutputFileNames({ jsFilePath, sourceMapFilePath, declarationFilePath}: { jsFilePath: string, sourceMapFilePath: string, declarationFilePath: string },

```

Comment:
should `isBundleEmit` and `sourceFiles` be optional in the definition of  `forEachExpectedEmitFile(host: EmitHost,
        action: (emitFileNames: EmitFileNames, sourceFiles: SourceFile[], isBundledEmit: boolean) => void,
        targetSourceFile?: SourceFile)`